### PR TITLE
RouterOS Packet Manager Plugin

### PR DESCRIPTION
**Pull request by csylos**
Plugin that turns a RouterOS device into an in-line packet manager, with queue sync and queue verification.

### CODE REVIEW

File: plugins/ros-pac-man/manifest.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+    "version": "1",
+    "information": {
+        "name": "pacman",
```
Comment (keksa) [inline plugins/ros-pac-man/manifest.json]: name of the plugin in manifest should correspond to the directory here, can you rename it?
I think ideal name would be `routeros-packet-manager` for both directory and manifest

File: plugins/ros-pac-man/manifest.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        "displayName": "RouterOS Packet Manager",
+        "description": "This plugin allows using Mikrotik as a packet manager",
+        "url": "http://www.atlinkservices.com",
```
Comment (keksa) [inline plugins/ros-pac-man/manifest.json]: The URL was originally meant in the way, that when user clicks it in UCRM, he goes to a page, where he can get more information about the plugin or plugin updates.

If you want to keep it this way, feel free. Although I think, that better way if you don't have standalone page on your website, would be to link the plugin directory in this repo and include this URL in the author field (e.g. "Shaun Wilkinson - www.atlinkservices.com").

File: plugins/ros-pac-man/src/ucrmInfo.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    
+    $pm2AdminPass = $configs['pm2AdminPass'];
+    
+    // Specific UCRM URL for private use
+    
+    $ucrmURL = $ucrmPublicURL."api/v1.0";
+    
+    // UCRM API key with read or write capability
+    
+    $ucrmKey = "YOUR_KEY_HERE";
```
Comment (keksa) [inline plugins/ros-pac-man/src/ucrmInfo.php]: you can get this (`pluginAppKey`) from the automatically created `ucrm.json` file in the root of plugin, take a look at docs for more info https://github.com/Ubiquiti-App/UCRM-plugins/blob/master/docs/file-structure.md#ucrmjson

this plugin is unusable as of now without editing the code

File: plugins/ros-pac-man/manifest.json (hunk 1 above)
Comment (csylos) [reply plugins/ros-pac-man/manifest.json]: Edited to reflect new package name.

File: plugins/ros-pac-man/manifest.json (hunk 2 above)
Comment (csylos) [reply plugins/ros-pac-man/manifest.json]: Edited to refer to plugin page.

File: plugins/ros-pac-man/src/ucrmInfo.php (hunk 3 above)
Comment (csylos) [reply plugins/ros-pac-man/src/ucrmInfo.php]: Edited to pull the key from ucrm.json.

**Comment by csylos** [comment]
All changes requested have been made.

File: plugins/ros-pac-man/manifest.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
+    "version": "1",
+    "information": {
+        "name": "routeros-packet-manager",
```
Comment (keksa) [inline plugins/ros-pac-man/manifest.json]: last thing to do is to rename the directory (`ros-pac-man` -> `routeros-packet-manager`)

other than that, everything looks fine

**Pull request closed without merging**